core/app/models/spree/stock/coordinator.rb

```diff
@@ -66,12 +65,19 @@ def build_packages(packages = Array.new)
 
       private
 
-      def unallocated_inventory_units
-        inventory_units - @preallocated_inventory_units
+      def stock_location_variant_ids
+        location_variant_ids = StockItem.where(variant_id: unallocated_variant_ids).joins(:stock_location).merge(StockLocation.active).pluck(:stock_location_id, :variant_id)
+
+        location_lookup = StockLocation.where(id: location_variant_ids.map(&:first).uniq).map { |l| [l.id, l] }.to_h
+
+        location_variant_ids.each_with_object({}) do |(location_id, variant_id), hash|
+          hash[location_lookup[location_id]] ||= Set.new
+          hash[location_lookup[location_id]] << variant_id
+        end
```

Comment:
@magnusvk I think `Enumerable#each_with_object` will return the `hash`. I believe the nature of it was so that you didn't have to store it in a variable and then return it :p